### PR TITLE
Change public API TPM2_Init_minimal to take user context

### PR DESCRIPTION
**Pull request by elms**

### CODE REVIEW

File: src/tpm2.c [hunk 1]
```diff
@@ -388,7 +388,7 @@ TPM_RC TPM2_Init_ex(TPM2_CTX* ctx, TPM2HalIoCb ioCb, void* userCtx,
     return rc;
 }
 
-TPM_RC TPM2_Init_minimal(TPM2_CTX* ctx)
+TPM_RC TPM2_Init_minimal(TPM2_CTX* ctx, void* userCtx)
```
Comment (dgarske) [inline src/tpm2.c:391]: Changes to public API's are not allowed. However this was just added 4/28/20 for the `--enable-devtpm` support and is unlikely to be an issue. The `TPM2_Init_minimal` is used by the wrapper only.

**Review by dgarske — APPROVED** [review]
Thanks Jeff. I like this change, but need to confirm breaking this public API is okay before merge.

File: src/tpm2.c (hunk 1 above)
Comment (elms) [reply src/tpm2.c:391]: I misunderstood how `userCtx` is used. My new understanding is that `userCtx` is only used when `ioCb` is called, which is only for `TIS` functions. When `TPM2_Init_minimal` is used by wrap, TIS is not used. @dgarske is that correct?

If so, then I should close this PR and instead open a PR with additional comments and not consider not even building TIS and erroring if `ioCb` or `userCtx` are anything besides NULL when `devtpm` or `swtpm` enabled.

Comment (dgarske) [reply src/tpm2.c:391]: That is correct. Agree we can close. Thanks

**Pull request closed without merging**